internal/runtime/compiler/checker/checker.go

```diff
@@ -838,7 +849,9 @@ func (c *checker) VisitAfter(node ast.Node) ast.Node {
 			return n
 		}
 		n.Pattern = pe.pattern.String()
-		c.checkRegex(n.Pattern, n)
+		if !c.skipRegexCheck {
```

Comment:
`checkRegex` also does a parse test that is now being skipped for subst arguments, so this weakens the pass, deferring regex syntax errors until code generation time.

I'd prefer this be passed as an argument, to disable the `capNames` loop on checker.go:894.

I'd actually prefer a better way of handling symbol insertion such that positional capref names are only checked if they're referenced later on.  This might mean adding a per-regexp scope and then promoting referenced identifiers out of that scope, which will be weird and require special logic in the symbol lookup.

Or... allow the symbol table to contain mutiple symbols under a name, and then throw the error in `checkSymbolTable()` if there is more than one assigned to a name.

Reply:
I see what you're saying and I've updated this and added tests to make sure this is still checked (and verified that before making the updates the tests were failing, which is what you were pointing out).

Cross referencing the grammar and the VM it looks like the only way `t.matches` is updated is if a `code.Match` or `code.Smatch` opcode is emitted which appears to be a `/pattern/ {}` or a `var [=!]~ /pattern/`, so it doesn't look like these symbols should even exist on the checker's symbol table.

I think the two other symbol table handling changes can still apply as further improvements to the code, but I'm not sure how often they occur since the two opcodes I listed seem like they would be creating new scopes and I'd have to understand the VM and how variables are looked up to really update the checker to match the VM.

Reply:
Yep I agree.  I had a think about this last night after I wrote the comment.  The meaning of the regular expression here is not one that implies variable references.  At best one might think that other parameters to subst could be reference names but there is no scope leaked outside the function parameters.

I will take the task of fixing the symbol table code.